src/Handler/MockHandler.php

```diff
@@ -87,6 +87,19 @@ function ($value) use ($request, $options) {
                 if ($this->onFulfilled) {
                     call_user_func($this->onFulfilled, $value);
                 }
+                if (isset($options['sink'])) {
+                    $contents = (string) $value->getBody();
+                    $sink = $options['sink'];
+
+                    if (is_resource($sink)) {
```

Comment:
Yeah I've started with that implementation, but found out that creating a new stream (Psr7\stream_for) had some issues. I use a tmpfile which deletes the file once the stream is closed, which happens on Stream::__destroy. Any suggestion how this can be fixed?